src/Store.js

```diff
@@ -259,10 +259,19 @@ class Store {
       return head
     }
 
-    return mapSeries(heads, saveToIpfs)
-      .then(async (saved) => {
-        return this._replicator.load(saved.filter(e => e !== null))
+    const saved = await mapSeries(heads, saveToIpfs)
+    await this._replicator.load(saved.filter(e => e !== null))
+
+    return new Promise(resolve => {
```

Comment:
I feel like this also needs a timeout to be rejected at some point

Reply:
Not sure I follow, why would you want to do that?

Reply:
`resolve` is wrapped in conditionals both of the times. Are we sure one of those is always going to happen?

Reply:
I'm pretty sure yes. If not something with the replication logic would be broken